fms/models/siglip_vision.py

```diff
@@ -0,0 +1,399 @@
+
+        self.embeddings = SiglipVisionEmbeddings(self.config)
+        self.encoder = SiglipEncoder(self.config)
+        self.post_layernorm = LayerNormParameterized(
+            self.config.hidden_size,
+            elementwise_shift=True,
+            use_mean=True,
+            eps=self.config.layer_norm_eps,
+            use_high_precision_pow=True,
+        )
+
+    def reset_parameters(self):
+        for m in self.modules():
+            m.reset_parameters()
+
+    def post_init(self):
+        self.embeddings.post_init()
+
+    def forward(
+        self,
+        pixel_values,
+        **attn_kwargs: Unpack[AttentionKwargs],
+    ):
+        hidden_states = self.embeddings(pixel_values)
+        hidden_states = self.encoder(inputs_embeds=hidden_states, **attn_kwargs)
+        hidden_states = self.post_layernorm(hidden_states)
+        return hidden_states
+
+
+class SiglipVision(nn.Module):
+    def __init__(
+        self,
+        config: Optional[SiglipVisionConfig] = None,
+        distributed_strategy: DistributedStrategy = NoOpStrategy,
+        **kwargs,
+    ):
+        super(SiglipVision, self).__init__()
+        if config is not None:
+            self.config = config
+        else:
+            self.config = SiglipVisionConfig()
+        self.config = self.config.updated(**kwargs)
+        self.distributed_strategy = distributed_strategy
+
+        self.base_model = SiglipVisionHeadless(self.config, self.distributed_strategy)
+
+        self.head = SiglipMultiheadAttentionPoolingHead(self.config)
+
+    @classmethod
+    def from_config(cls, config: SiglipVisionConfig) -> "SiglipVision":
+        return cls(config)
+
+    def get_config(self) -> SiglipVisionConfig:
+        return self.config
+
+    def reset_parameters(self):
+        self.head.reset_parameters()
+        self.base_model.reset_parameters()
+
+    def post_init(self):
+        self.base_model.post_init()
+
+    def forward(
+        self,
+        pixel_values,
+        **attn_kwargs: Unpack[AttentionKwargs],
+    ):
+        hidden_states = self.base_model(pixel_values, **attn_kwargs)
+        pooler_output = self.head(hidden_states)
+        return hidden_states, pooler_output
+
+
+_siglip_base_patch16_224_config = SiglipVisionConfig()
+
+_architecture_name = "siglip_vision"
+
+
+def _siglip_vision_factory_factory(config):
+    def factory(**kwargs):
+        return SiglipVision(config, **kwargs)
+
+    return factory
+
+
+models.register_model(
+    _architecture_name,
+    "siglip_base_patch16_224",
+    _siglip_vision_factory_factory(_siglip_base_patch16_224_config),
+)
+
+
+def _weight_fusion(
+    input_sd: Mapping, model_config: Optional[SiglipVisionConfig] = None, **kwargs
+):
+    has_fused_weights = True
+    if model_config:
+        if not model_config.fused_weights:
+            has_fused_weights = False
+
+    new_sd = input_sd
+    if has_fused_weights:
+        new_sd = serialization._mlp_glu_unfused_to_fused_adapter_step(
```

Comment:
the MLP layer sin't GLU, so weight fusion with this helper won't work, there sohuld be another one that fuses the FCN instead

Reply:
Thanks! Will fix

Reply:
@ani300 There doesn't seem to be one specifically for mlp in utils/serialization.py

Reply:
oh nvm, that's what happens when you review code after 11pm. There's no fusion possible for MLP. Just remove the glu portion here, the code should be:

```python
    new_sd = input_sd
    if has_fused_weights:
        new_sd = serialization._attn_unfused_to_fused_step(new_sd)
    return new_sd
```

Reply:
Np! Thank for reviewing!